src/proto/convert.rs

```
@@ -81,3 +81,30 @@ impl From<JsonError> for Response {
 		}
 	}
 }
+
+impl From<QueryParseError> for Response {
+	fn from(error: QueryParseError) -> Response {
+		match error {
+			QueryParseError::None => {
+				let json = json!({
+					"error": "query data was empty",
+				});
+				res().with_body(to_string(&json).unwrap())
+			}
+
+			QueryParseError::ParseError(_e) => {
+				let json = json!({
+					"error": "failed to parse query data",
+				});
+				res().with_body(to_string(&json).unwrap())
+			}
+
+			QueryParseError::SerdeError(_e) => {
+				let json = json!({
+					"error": "failed to deserialize data",
+				});
+				res().with_body(to_string(&json).unwrap())
+			}
```

same here we should report the serde error to the user.